android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapFragment.java

```diff
@@ -287,35 +289,37 @@ private void updateOverlayBaseLayer(int zoomLevel) {
         final List<Overlay> overlays = mMap.getOverlays();
         final Overlay overlayRemoved = (!isHighZoom(zoomLevel)) ? mLowResMapOverlayHighZoom : mLowResMapOverlayLowZoom;
         final Overlay overlayAdded = (isHighZoom(zoomLevel)) ? mLowResMapOverlayHighZoom : mLowResMapOverlayLowZoom;
-        if (overlays.indexOf(overlayRemoved) > -1) {
-            overlays.remove(overlayRemoved);
-        }
-        if (overlays.indexOf(overlayAdded) == -1) {
+        overlays.remove(overlayRemoved);
+        if (!overlays.contains(overlayAdded)) {
             overlays.add(0, overlayAdded);
             mMap.invalidate();
         }
     }
 
-    // The MLS coverage follows the same logic as the lower resolution map overlay, in that
-    // when at low zoom level, show even lower resolution tiles. The MLS coverage is not
-    // dependant on the isHighBandwidth for this behaviour, it always does this.
+    // At regular zoom levels, the MLS coverage follows the same logic as the lower resolution map
+    // overlay, in that when at low zoom level, show even lower resolution tiles.
+    // At lower zoom levels (only available in unlimited zoom mode) show exact zoom level
     private void updateOverlayCoverageLayer(int zoomLevel) {
-        if (mCoverageTilesOverlayLowZoom == null || mCoverageTilesOverlayHighZoom == null) {
-            return;
-        }
-
         final List<Overlay> overlays = mMap.getOverlays();
-        int idx = 0;
-        if (overlays.indexOf(mLowResMapOverlayHighZoom) > -1 || overlays.indexOf(mLowResMapOverlayLowZoom) > -1) {
-            idx = 1;
+        if (mCoverageTilesOverlayLowZoom == null || mCoverageTilesOverlayHighZoom == null
```

Comment:
The precondition of this function should be the first line:
if (!isCoverageTilesInitialized()) {
  return;
}